pkg/util/sliding_window.go

```diff
@@ -83,6 +86,7 @@ func NewSlidingWindow(windowSize time.Duration, pollingInterval time.Duration) (
 		numBuckets:      int(windowSize / pollingInterval),
 		pollingInterval: pollingInterval,
 		windowSize:      windowSize,
+		clock:           clock.New(),
```

Comment:
Instead of instantiating the clock at this line, what do you think about injecting it (dependency injection)? You could have `NewSlidingWindowWithClock` that takes a clock as parameter and `NewSlidingWindow` that calls `NewSlidingWindowWithClock`. 
Doing so avoid the unit test to assume `NewSlidingWindow` returns an instance of  `slidingWindow` having a `clock` field.

Reply:
My thinking was to add as little non-`_test.go` code as possible to support this injection, but I'll switch it around to this method.